crates/consensus/common/src/validation.rs

```diff
@@ -300,33 +297,33 @@ pub fn validate_header_regarding_parent(
         return Err(ConsensusError::ParentBlockNumberMismatch {
             parent_block_number: parent.number,
             block_number: child.number,
-        });
+        })
     }
 
     if parent.hash != child.parent_hash {
         return Err(ConsensusError::ParentHashMismatch {
             expected_parent_hash: parent.hash,
             got_parent_hash: child.parent_hash,
-        });
+        })
     }
 
     // timestamp in past check
     if child.timestamp <= parent.timestamp {
         return Err(ConsensusError::TimestampIsInPast {
             parent_timestamp: parent.timestamp,
             timestamp: child.timestamp,
-        });
+        })
     }
 
     // TODO Check difficulty increment between parent and child
     // Ace age did increment it by some formula that we need to follow.
 
-    #[cfg(not(feature = "optimism"))]
-    check_gas_limit(parent, child, chain_spec)?;
-
-    #[cfg(feature = "optimism")]
-    if !chain_spec.optimism {
-        check_gas_limit(parent, child, chain_spec)?;
+    cfg_if::cfg_if! {
+        if #[cfg(feature = "optimism")] {
+            // On Optimism, the gas limit can adjust instantly, so we skip this check
+        } else {
+            check_gas_limit(parent, child, chain_spec)?;
+        }
```

Comment:
We want to still check if the chain is `optimism` even if reth is compiled with the `optimism` feature flag on.

Think this should be
```suggestion
    cfg_if::cfg_if! {
        if #[cfg(feature = "optimism")] {
            // On Optimism, the gas limit can adjust instantly, so we skip this check
            if !chain_spec.is_optimism() {
                check_gas_limit(parent, child, chain_spec)?;
            }
        } else {
            check_gas_limit(parent, child, chain_spec)?;
        }
```